test/app/components/Settings/scheduler/create-modal.test.jsx

```diff
@@ -110,4 +115,47 @@ describe('Create Modal Test', () => {
             cronInterval
         });
     });
+
+    it('should display backend error to user on failures', async () => {
+        const onSubmit = jest.fn(() => Promise.reject(new Error('Oops')));
+        const component = mount(
+            <CreateModal
+                open={true}
+                initialCode="SELECT * FROM foods"
+                initialFilename="filename"
+                onSubmit={onSubmit}
+            />
+        );
+
+        component
+            .find('button')
+            .at(1)
+            .simulate('click');
+
+        // onSubmit is asynchronous
+        await wait();
```

Comment:
if i remember correctly, our webpack/babel setup isn't ready to use `await`.

Reply:
Hmm. This worked for me though. Should i remove it and use `setTimeout` instead?

Reply:
Please, ignore my comment. It doesn't apply any longer. We upgrade to Node 8 recently.